services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/DirectionsRoute.java

```diff
@@ -110,6 +110,19 @@ public static Builder builder() {
   @Nullable
   public abstract List<RouteLeg> legs();
 
+  /**
+   * List of {@link DirectionsWaypoint} objects. Each {@code waypoint} is an input coordinate
+   * snapped to the road and path network. The {@code waypoint} appear in the list in the order of
+   * the input coordinates.
+   * Waypoints are returned in the {@link DirectionsRoute} object only when
+   * {@link RouteOptions#waypointsPerRoute()} is set to true. Otherwise they are returned
+   * in the root: {@link DirectionsResponse#waypoints()}.
+   *
+   * @return list of {@link DirectionsWaypoint} objects ordered from start of route till the end
+   */
+  @Nullable
+  public abstract List<DirectionsWaypoint> waypoints();
```

Comment:
But the parameter name is "waypoints". And you invoke it either on the route object or on the response. 
What's more, we've discussed that native waypoints will be the source of truth in the SDK.

Reply:
>But the parameter name is "waypoints"

we can use annotation `@SerializedName` 😃 

>And you invoke it either on the route object or on the response.
What's more, we've discussed that native waypoints will be the source of truth in the SDK.

the SDK is not the only customer of the mapbox-java, it might be used without the SDK. 

I'm just worrying that signatures are really similar and even if you know where your data is, you might make a mistake

Reply:
> we can use annotation @SerializedName 😃

I know but I'm just not sure we should use it in this situation and duplicate the knowledge that it's a route waypoint.

> I'm just worrying that signatures are really similar and even if you know where your data is, you might make a mistake

But we also have distance and duration for `DirectionsRoute`, `RouteLeg` and `LegStep`.

Reply:
ok, it's not a blocker, just a concern. I would hear other opinions, but if there are no any I'm good to keep it as is 👍